### PR TITLE
Replace `linked_list_allocator` with `rlsf` in the enclave binaries

### PR DESCRIPTION
**Pull request by andrisaar**
The malloc algorithm provided by `rlsf` is much more efficient than the one provided by `linked_list_allocator`.

How much more efficient? With a certain pathological data set it took 11 hours (!) to parse the protobuf containing the data. With `rlsf`, it took 11 seconds. Yes, seconds.

### CODE REVIEW

**Review by conradgrobler — APPROVED** [review]
NIce, thanks.

File: oak_enclave_runtime_support/src/heap.rs [hunk 1]
```diff
@@ -16,14 +16,47 @@
...
+
+impl Source {
+    // Restricted Kernel deals in 2 MiB pages, so that's what we use to request memory.
+    const PAGE_SIZE: usize = 0x20_0000usize;
+}
+
+unsafe impl FlexSource for Source {
+    unsafe fn alloc(&mut self, min_size: usize) -> Option<NonNull<[u8]>> {
+        // Ensure that we're allocating page-sized chunks of memory.
+        let size = if min_size % Self::PAGE_SIZE != 0 {
```
Comment (mariaschett) [inline oak_enclave_runtime_support/src/heap.rs:36]: To double check my understanding: you could also:

```
let missing_size_for_alignmet = min_size % Self::PAGE_SIZE;
let size = min_size + missing_size_for_alignment;
```

**Review by mariaschett — APPROVED** [review]
Thanks!

File: oak_enclave_runtime_support/src/heap.rs (hunk 1 above)
Comment (andrisaar) [reply oak_enclave_runtime_support/src/heap.rs:36]: Close, but `missing_size` would have to be `Self::PAGE_SIZE - min_size % Self::PAGE_SIZE`.

...which is actually much simpler so I'll switch to that.

Comment (andrisaar) [reply oak_enclave_runtime_support/src/heap.rs:36]: Actually no, doing it like that would increase the size by one page if `size` is already page-aligned. We don't want that.

**Pull request merged**